### PR TITLE
[Android] Verify correct behavior of bleed property

### PR DESCRIPTION
**Pull request by almedina-ms**
Updated bleed behavior for android and consolidated some logic into functions so the padding and bleed logic would not be repeated 

### How verified 
The mobile app was run and these are the results

### Comprehensive payload
Screenshot_19700120-212829
Screenshot_19700120-212834
Screenshot_19700120-212841

### Column.Bleed
Screenshot_19700120-212846

### ColumnSet.Bleed
Screenshot_19700120-212852

### Container.Bleed
Screenshot_19700120-212857

### CODE REVIEW

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java [hunk 1]
```diff
@@ -167,5 +158,45 @@ else if ((container = Container.dynamic_cast(baseCardElement)) == null)
...
+            LinearLayout.LayoutParams layoutParams = (LinearLayout.LayoutParams) collectionElementView.getLayoutParams();
+            int marginLeft = layoutParams.leftMargin, marginRight = layoutParams.rightMargin;
+
+            ContainerBleedDirection bleedDirection = collectionElement.GetBleedDirection();
+            if (bleedDirection == ContainerBleedDirection.BleedToLeading || bleedDirection == ContainerBleedDirection.BleedToBothEdges)
+            {
+                marginLeft = -padding;
+            }
+
+            if(bleedDirection == ContainerBleedDirection.BleedToTrailing || bleedDirection == ContainerBleedDirection.BleedToBothEdges)
```
Comment (paulcam206) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java]: nit: space after `if`

**Comment by dclaux** [comment]
Per the screenshots, there seems to still be a couple things not handled as expected. Namely, bleeding at the top/bottom of containers/the card itself doesn't seem to be handled. A container that bleeds should bleed not only left and right, but also top and bottom when appropriate. In the first screenshot for instance, the "Bleeding container at the top of the card" should bleed into the top padding of the card.

I am going to assume that this has been compared to the Designer's rendering; the designer voluntarily disables bleeding top and bottom in order to leave space at the top and bottom of any container onto which elements can be dragged (otherwise, it would be impossible to reorder elements.) To see final render of the card, use the "Preview mode" button. Or, use the visualizer instead.

**Comment by RebeccaAnne** [comment]
We don't currently have that behavior on any of our non-javascript renderers, as we were all using the designer as a comparison point, and it's not super explicit in the spec. It does look like the last gif example shows that behavior (I assume "marginFromParent":"none" is an old way to say bleed?), but looks like we missed it. I'll log a bug to track addressing it separate of this PR.

---
In reply to: [484155715](https://github.com/Microsoft/AdaptiveCards/pull/2700#issuecomment-484155715) [](ancestors = 484155715)

**Comment by RebeccaAnne** [comment]
#2702

Also, i understand the concern about needing the visualizer to be usable, but are we concerned that users won't realize that what they see in the visualizer is not what will ultimately be rendered? (Also, where is the "Preview mode" button? I can't seem to find it.)

---
In reply to: [484190751](https://github.com/Microsoft/AdaptiveCards/pull/2700#issuecomment-484190751) [](ancestors = 484190751,484155715)

**Comment by dclaux** [comment]
I believe the test payload you guys used is from here: https://github.com/Microsoft/AdaptiveCards/issues/2109

There is also a screenshot of the rendered result in there; that is the reference.

**Comment by dclaux** [comment]
Yes, there is a risk of potential confusion, and I have ideas for making it better. However these improvements will come later, and in general the designer should NOT be used as a reference for the final visuals (unless using the "Preview mode" feature in the 2.0 early preview) because the designer will always behave slightly differently in some cases to facilitate drag and drop.

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java [hunk 2]
```diff
@@ -163,5 +154,45 @@ else if ((container = Container.dynamic_cast(baseCardElement)) == null)
...
+            }
+
+            layoutParams.setMargins(marginLeft, layoutParams.topMargin, marginRight, layoutParams.bottomMargin);
+            collectionElementView.setLayoutParams(layoutParams);
+        }
+    }
+
+    public static void ApplyPadding(ContainerStyle elementContainerStyle, ContainerStyle parentContainerStyle, LinearLayout collectionElementView, Context context, HostConfig hostConfig)
+    {
+        if (elementContainerStyle != parentContainerStyle)
```
Comment (nesalang) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java]: just want to let you know that you can call GetPadding() on collectionType object to see if it gets padding.

**Pull request merged**